docs/book/user-guide/production-guide/ci-cd.md

```diff
@@ -6,28 +6,94 @@ description: >-
+## Configure an API Key in ZenML
 
-Once the data scientist thinks they have improved the pipeline, they create a pull request for their branch on GitHub. This automatically triggers a GitHub Action that will run the same pipeline in the staging environment (e.g. a pipeline running on a cloud stack in GCP), potentially with different test data. As long as the pipeline does not run successfully in the staging environment, the PR cannot be merged. The pipeline also generates a set of metrics and test results that are automatically published to the PR, where they can be peer-reviewed to decide if the changes should be merged.
+In order to facilitate machine-to-machine connection you need to create an API key within ZenML. Learn more about those
+[here](https://docs.zenml.io/how-to/connecting-to-zenml/connect-with-a-service-account).
 
-![Pipeline with staging stack](../../.gitbook/assets/ci-cd-staging.png)
+```bash
+zenml service-account create github_action_api_key
+```
 
-Once the PR has been reviewed and passes all checks, the branch is merged into `main`. This automatically triggers another GitHub Action that now runs a pipeline in the production environment, which trains the same model on production data, runs some checks to compare its performance with the model currently served in production and then, if all checks pass, automatically deploys the new model.
+This will return the API Key to you like this. This will not be shown to you again, so make sure to copy it here for
+use in the next section.
 
-![Pipeline with production stack](../../.gitbook/assets/ci-cd-prod.png)
+```bash
+Created service account 'github_action_api_key'.
+Successfully created API key `default`.
+The API key value is: 'ZENKEY_...'
+Please store it safely as it will not be shown again.
+To configure a ZenML client to use this API key, run:
 
-To see this in action, check out the [ZenML Gitflow Repository](https://github.com/zenml-io/zenml-gitflow/). This repository showcases how ZenML can be used for machine learning with a GitHub workflow that automates CI/CD with continuous model training and continuous model deployment to production. The repository is also meant to be used as a template: you can fork it and easily adapt it to your own MLOps stack, infrastructure, code and data.
+...
+```
+
+## Set up your secrets in Github
+
+For our Github Actions we will need to set up some secrets [for our repository](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions#creating-secrets-for-a-repository). 
+Specifically, you should use github secrets to store the `ZENML_API_KEY` that you created above.
+
+![create_gh_secret.png](../../.gitbook/assets/create_gh_secret.png)
+
+The other values that are loaded from secrets into the environment [here](https://github.com/zenml-io/zenml-gitflow/blob/main/.github/workflows/pipeline_run.yaml#L14-L23)
+can also be set explicitly or as variables.
+
+## Set up different stacks for Staging and Production
```

Comment:
its not just stacks but you might want to parameterize the data loading steps to read different things on staging and prod